### ISSUE TITLE
Clangd over SSH from a windows machine

### ISSUE THREAD

**Issue by qchateau** [clangd/vscode-clangd#133]
When using VSCode on a windows machine, connecting to a linux machine over SSH (with the SSH remote extension of VSCode), clangd cannot start if arguments are used.

I worked around the problem by creating a bash script on my remote machine, which calls clangd with the arguments I want. Then I use this bash script instead of clangd as the target executable of the VSCode extension.

No logs are available in vscode as clangd does not even start.

I'll try to find some time to investigate further but maybe there are people on this repo who are comfortable with this extension's code that can figure out what's wrong in the argument handling.

**Comment by qchateau** [comment]
Hu, could not reproduce. Will reopen if I can

**Comment by yuedai** [comment]
I meet same issue. Is this a configuration issue?

**Comment by yuedai** [comment]
Do I need to install llvm/clang in my Windows desktop? I only have LLVM deployed on remote CentOS7. The LLVM is not in standard path. Instead it's located in $PATH by .bashrc file.

**Comment by femto-code** [comment]
@qchateau 
Could you explain how you have accomplished the workaround via the bash script?
I cannot get the clangd extension to work on remote SSH machine...

**Comment by qchateau** [comment]
I just created an executable bash script running clangd on the remote machine and changed the path to the clangd binary to run it instead of clangd. Note that I don't have this problem anymore 

```bash
#!/bin/bash

clangd
```

**Comment by epistax** [comment]
> I just created an executable bash script running clangd on the remote machine and changed the path to the clangd binary to run it instead of clangd. Note that I don't have this problem anymore
> 
> #!/bin/bash
> 
> clangd

Thank you for this. Three years later, using vscode 1.98.2 and llvm 20.1.1, this problem still exists and this workaround still works. In my case clangd is also not on the path but it's within the workspace. However just telling it to invoke the file in the workspace results in an EACCES error when using vscode via ssh. When running vscode locally I don't have the same issue. I dropped a clangd.sh into the root of my workspace, I path to the version of clangd I want it to run in that file, and I pointed vscode to that .sh file and it works fine now.
